src/WebJobs.Script.WebHost/Security/KeyManagement/KubernetesSecretsRepository.cs

```diff
@@ -0,0 +1,216 @@
+                }
+
+                var functionSecret = secrets as FunctionSecrets;
+                if (functionSecret?.Keys != null)
+                {
+                    foreach (var key in functionSecret.Keys)
+                    {
+                        newKeys[$"{FunctionKeyPrefix}{functionName}.{key.Name}"] = key.Value;
+                    }
+                }
+            }
+            return newKeys;
+        }
+
+        /// <summary>
+        /// no-op - allow stale secrets to remain
+        /// </summary>
+        public async Task PurgeOldSecretsAsync(IList<string> currentFunctions, ILogger logger)
+            => await Task.Yield();
+
+        /// <summary>
+        /// Runtime is not responsible for encryption so this code will never be executed.
+        /// </summary>
+        public Task WriteSnapshotAsync(ScriptSecretsType type, string functionName, ScriptSecrets secrets)
+            => throw new NotSupportedException();
+
+        /// <summary>
+        /// Runtime is not responsible for encryption so this code will never be executed.
+        /// </summary>
+        public Task<string[]> GetSecretSnapshots(ScriptSecretsType type, string functionName)
+            => throw new NotSupportedException();
+
+        private async Task<ScriptSecrets> ReadHostSecrets()
+        {
+            IDictionary<string, string> secrets = await _kubernetesClient.GetSecrets();
+
+            HostSecrets hostSecrets = new HostSecrets()
+            {
+                FunctionKeys = new List<Key>(),
+                SystemKeys = new List<Key>()
+            };
+
+            foreach (var pair in secrets)
+            {
+                if (pair.Key.StartsWith(MasterKey))
+                {
+                    hostSecrets.MasterKey = new Key("master", pair.Value);
+                }
+                else if (pair.Key.StartsWith(HostFunctionKeyPrefix))
+                {
+                    hostSecrets.FunctionKeys.Add(ParseKeyWithPrefix(HostFunctionKeyPrefix, pair.Key, pair.Value));
+                }
+                else if (pair.Key.StartsWith(SystemKeyPrefix))
+                {
+                    hostSecrets.SystemKeys.Add(ParseKeyWithPrefix(SystemKeyPrefix, pair.Key, pair.Value));
+                }
+            }
+
+            return hostSecrets?.MasterKey == null ? null : hostSecrets;
+        }
+
+        private async Task<ScriptSecrets> ReadFunctionSecrets(string functionName)
+        {
+            IDictionary<string, string> secrets = await _kubernetesClient.GetSecrets();
+            var prefix = $"{FunctionKeyPrefix}{functionName}.";
+
+            var functionSecrets = new FunctionSecrets()
+            {
+                Keys = secrets
+                    .Where(p => p.Key.StartsWith(prefix))
+                    .Select(p => ParseKeyWithPrefix(prefix, p.Key, p.Value))
+                    .ToList()
+            };
+
+            return functionSecrets.Keys.Count == 0 ? null : functionSecrets;
```

Comment:
same thing here. as with the HostSecrets above and assumptions about null vs !null from the SecretsManager